iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/util/RegulatedTerminalControllers.java

```diff
@@ -0,0 +1,223 @@
+                    replaceRegulatedTerminalThreeWindingsTransformer((ThreeWindingsTransformer) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            case GENERATOR ->
+                    replaceRegulatedTerminalGenerator((Generator) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            case SHUNT_COMPENSATOR ->
+                    replaceRegulatedTerminalShuntCompensator((ShuntCompensator) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            case STATIC_VAR_COMPENSATOR ->
+                    replaceRegulatedTerminalStaticVarCompensator((StaticVarCompensator) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            case HVDC_CONVERTER_STATION ->
+                    replaceRegulatedTerminalHvdcConverterStation((HvdcConverterStation<?>) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            case BATTERY ->
+                    replaceRegulatedTerminalBattery((Battery) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            case VOLTAGE_LEVEL ->
+                    replaceRegulatedTerminalVoltageLevel((VoltageLevel) identifiable, currentRegulatedTerminal, newRegulatedTerminal);
+            default -> throw new PowsyblException("unexpected identifiable type: " + identifiable.getType());
+        }
+    }
+
+    private static void replaceRegulatedTerminalTwoWindingsTransformer(TwoWindingsTransformer t2w, Terminal currentRegulatedTerminal, Terminal newRegulatedTerminal) {
+        t2w.getOptionalRatioTapChanger().ifPresent(rtc -> replace(rtc, currentRegulatedTerminal, newRegulatedTerminal));
+        t2w.getOptionalPhaseTapChanger().ifPresent(ptc -> replace(ptc, currentRegulatedTerminal, newRegulatedTerminal));
+    }
+
+    private static void replaceRegulatedTerminalThreeWindingsTransformer(ThreeWindingsTransformer t3w, Terminal currentRegulatedTerminal, Terminal newRegulatedTerminal) {
+        t3w.getLeg1().getOptionalRatioTapChanger().ifPresent(rtc -> replace(rtc, currentRegulatedTerminal, newRegulatedTerminal));
+        t3w.getLeg1().getOptionalPhaseTapChanger().ifPresent(ptc -> replace(ptc, currentRegulatedTerminal, newRegulatedTerminal));
+        t3w.getLeg2().getOptionalRatioTapChanger().ifPresent(rtc -> replace(rtc, currentRegulatedTerminal, newRegulatedTerminal));
+        t3w.getLeg2().getOptionalPhaseTapChanger().ifPresent(ptc -> replace(ptc, currentRegulatedTerminal, newRegulatedTerminal));
+        t3w.getLeg3().getOptionalRatioTapChanger().ifPresent(rtc -> replace(rtc, currentRegulatedTerminal, newRegulatedTerminal));
+        t3w.getLeg3().getOptionalPhaseTapChanger().ifPresent(ptc -> replace(ptc, currentRegulatedTerminal, newRegulatedTerminal));
+    }
+
+    private static void replace(TapChanger<?, ?, ?, ?> tc, Terminal currentRegulatedTerminal, Terminal newRegulatedTerminal) {
+        if (tc.getRegulationTerminal().equals(currentRegulatedTerminal)) {
```

Comment:
Same thing, terminal equality should not be tested using `.equals`:
- change all the `.equals(currentRegulatedTerminal)` tests in this method and all the following `replace...()` methods by tests checking the terminal connectable ids.

Reply:
Done